ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -1336,6 +1341,9 @@ public static long getSizeOfPrimitiveTypeArraysFromType(String colType, int leng
    */
   public static long getSizeOfMap(StandardConstantMapObjectInspector scmoi) {
     Map<?, ?> map = scmoi.getWritableConstantValue();
+    if (null == map || map.isEmpty()) {
+      return 0L;
+    }
```

Comment:
This changes the behavior of the existing code.

An empty map may return a value greater than 0 further down in this method:

```
// A result may be > 0 because of the map's overhead
result += JavaDataModel.get().hashMap(map.entrySet().size());
```

Reply:
Thanks for your review, maybe we can return it like this:
```
Map<?, ?> map = scmoi.getWritableConstantValue();
    if (null == map || map.isEmpty()) {
      return JavaDataModel.get().hashMap(0);
    }
```

Reply:
I'm actually not sure what the correct behavior is here for a 'null' Map.  I'd have to dig in more to see if there is already precedence for that.  I just know that the current proposal is not correct.  Maybe someone else can weigh in on the correct handling of a null value.

Reply:
We can ask the original author's opinion, maybe he has a better idea. 
Hi @prasanthj , Could you help me review it ?

Reply:
I think it will be ok

calculating with the size of an empty hashmap is an overestimation of the space needed to represent `null` - so we are making mistakes in the right direction....

Reply:
@kgyrtkirk I think a 'null' map can be size zero.  But any empty map is different and should probably include the overheard size (and not change the current behavior as part of this JIRA).

Reply:
wasn't the old behaviour an exception? :D
I think anything is better than that...for null it can be `0` as well; but in any case for empty stuff it may never be 0

Reply:
@kgyrtkirk As I understand it, the issue was with 'null' value, not empty maps.  The code in master branch returns a value (probably 4) to account for the overhead of having a map, even if it's empty.

What I am proposing is:

```
    if (null == map) {
      return 0L;
    }
   // otherwise, do the other/old/current stuff...
```

Reply:
oh, sorry!...
of course; that perfectly makes sense!

Reply:
@StefanXiepj could you update that if according @belugabehr 's proposal?

Reply:
@belugabehr & @kgyrtkirk , I agree entirely with you!  It have been updated.